radio/src/maths.cpp

```diff
@@ -172,3 +173,30 @@ void getGpsDistance()
     telemetryData.hub.maxGpsDistance = telemetryData.hub.gpsDistance;
 }
 #endif
+
+#if defined(STM32F4)
+uint32_t getCoordDistance(float  y1, float x1, float y2, float x2){
+  float nRadius = 6370997.0; // Earth’s radius in meters
+  float rad = 0.0174532925199433;
+  float nDLat = (y2 - y1) * rad;
+  float nDLon = (x2 - x1) * rad;
+  y1 *= rad;
+  y2 *= rad;
+
+  float nA =   powf( sinf(nDLat/2), 2 ) + cosf(y1) * cosf(y2) * powf( sin(nDLon/2), 2 );
```

Comment:
I'd get rid of the `powf` calls here.  Calculate the sin values and store in a variable, then multiply those together, it should save a few CPU cycles.  Also you probably meant to use `sinf` in the last instance.